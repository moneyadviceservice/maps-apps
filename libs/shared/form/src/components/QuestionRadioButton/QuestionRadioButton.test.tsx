import { render, screen } from '@testing-library/react';

import { QuestionRadioButton } from '.';

describe('Question Radio Button component', () => {
  it('renders correctly', () => {
    render(
      <QuestionRadioButton
        name={'test'}
        options={[
          { text: 'Option 1', value: 'option1' },
          { text: 'Option 2', value: 'option2' },
        ]}
      >
        Test
      </QuestionRadioButton>,
    );
    const question = screen.getByTestId('question-radio');
    expect(question).toMatchSnapshot();
  });

  it('renders with horizontal layout', () => {
    render(
      <QuestionRadioButton
        name={'test'}
        options={[
          { text: 'Option 1', value: 'option1' },
          { text: 'Option 2', value: 'option2' },
        ]}
        horizontalLayout={true}
      >
        Test
      </QuestionRadioButton>,
    );
    const question = screen.getByTestId('question-radio');
    expect(question).toMatchSnapshot();
  });

  it('renders with error state', () => {
    render(
      <QuestionRadioButton
        name={'test'}
        options={[
          { text: 'Option 1', value: 'option1' },
          { text: 'Option 2', value: 'option2' },
        ]}
        hasError={true}
      >
        Test
      </QuestionRadioButton>,
    );
    const question = screen.getByTestId('question-radio');
    expect(question).toMatchSnapshot();
  });

  it('renders with no children', () => {
    render(
      <QuestionRadioButton
        name={'test'}
        options={[
          { text: 'Option 1', value: 'option1' },
          { text: 'Option 2', value: 'option2' },
        ]}
      />,
    );
    const question = screen.getByTestId('question-radio');
    expect(question).toMatchSnapshot();
  });

  it('renders with disabled options', () => {
    render(
      <QuestionRadioButton
        name={'test'}
        options={[
          { text: 'Option 1', value: 'option1', disabled: true },
          { text: 'Option 2', value: 'option2' },
        ]}
      >
        Test
      </QuestionRadioButton>,
    );
    const question = screen.getByTestId('question-radio');
    expect(question).toMatchSnapshot();
  });

  it('renders with default checked option', () => {
    render(
      <QuestionRadioButton
        name={'test'}
        options={[
          { text: 'Option 1', value: 'option1' },
          { text: 'Option 2', value: 'option2' },
        ]}
        defaultChecked={'option1'}
      >
        Test
      </QuestionRadioButton>,
    );
    const question = screen.getByTestId('question-radio');
    expect(question).toMatchSnapshot();
  });
});
