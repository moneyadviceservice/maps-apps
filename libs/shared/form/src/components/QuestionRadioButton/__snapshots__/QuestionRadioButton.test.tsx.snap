// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Radio Button component renders correctly 1`] = `
<div
  data-testid="question-radio"
>
  <legend
    class="mb-2 text-lg font-bold text-gray-800"
  >
    Test
  </legend>
  <div
    class="flex flex-col gap-4 mt-3"
  >
    <div
      class=""
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          id="test-0"
          name="test"
          required=""
          type="radio"
          value="option1"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100"
          for="test-0"
        >
          Option 1
        </label>
      </div>
    </div>
    <div
      class=""
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          id="test-1"
          name="test"
          required=""
          type="radio"
          value="option2"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100"
          for="test-1"
        >
          Option 2
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Question Radio Button component renders with default checked option 1`] = `
<div
  data-testid="question-radio"
>
  <legend
    class="mb-2 text-lg font-bold text-gray-800"
  >
    Test
  </legend>
  <div
    class="flex flex-col gap-4 mt-3"
  >
    <div
      class=""
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          checked=""
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          id="test-0"
          name="test"
          required=""
          type="radio"
          value="option1"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100"
          for="test-0"
        >
          Option 1
        </label>
      </div>
    </div>
    <div
      class=""
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          id="test-1"
          name="test"
          required=""
          type="radio"
          value="option2"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100"
          for="test-1"
        >
          Option 2
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Question Radio Button component renders with disabled options 1`] = `
<div
  data-testid="question-radio"
>
  <legend
    class="mb-2 text-lg font-bold text-gray-800"
  >
    Test
  </legend>
  <div
    class="flex flex-col gap-4 mt-3"
  >
    <div
      class="opacity-25"
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          disabled=""
          id="test-0"
          name="test"
          required=""
          type="radio"
          value="option1"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:w-0 after:h-0 after:border-solid after:rounded-full peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100 opacity-60 cursor-not-allowed after:top-[1px] after:left-[1px] after:border-[19px] after:border-gray-200 after:opacity-100 hover:after:border-gray-200"
          for="test-0"
        >
          Option 1
        </label>
      </div>
    </div>
    <div
      class=""
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          id="test-1"
          name="test"
          required=""
          type="radio"
          value="option2"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100"
          for="test-1"
        >
          Option 2
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Question Radio Button component renders with error state 1`] = `
<div
  data-testid="question-radio"
>
  <legend
    class="mb-2 text-lg font-bold text-gray-800"
  >
    Test
  </legend>
  <div
    class="flex flex-col gap-4 mt-3"
  >
    <div
      class=""
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer before:border-2 before:border-red-700"
          id="test-0"
          name="test"
          required=""
          type="radio"
          value="option1"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100 before:border-2 before:border-red-700"
          for="test-0"
        >
          Option 1
        </label>
      </div>
    </div>
    <div
      class=""
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer before:border-2 before:border-red-700"
          id="test-1"
          name="test"
          required=""
          type="radio"
          value="option2"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100 before:border-2 before:border-red-700"
          for="test-1"
        >
          Option 2
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Question Radio Button component renders with horizontal layout 1`] = `
<div
  data-testid="question-radio"
>
  <legend
    class="mb-2 text-lg font-bold text-gray-800"
  >
    Test
  </legend>
  <div
    class="flex gap-4 mt-3 flex-row"
  >
    <div
      class=""
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          id="test-0"
          name="test"
          required=""
          type="radio"
          value="option1"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100"
          for="test-0"
        >
          Option 1
        </label>
      </div>
    </div>
    <div
      class=""
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          id="test-1"
          name="test"
          required=""
          type="radio"
          value="option2"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100"
          for="test-1"
        >
          Option 2
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Question Radio Button component renders with no children 1`] = `
<div
  data-testid="question-radio"
>
  <div
    class="flex flex-col gap-4 mt-3"
  >
    <div
      class=""
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          id="test-0"
          name="test"
          required=""
          type="radio"
          value="option1"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100"
          for="test-0"
        >
          Option 1
        </label>
      </div>
    </div>
    <div
      class=""
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          id="test-1"
          name="test"
          required=""
          type="radio"
          value="option2"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-checked:hover:after:border-pink-800 peer-focus:hover:after:border-magenta-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-pink-400 peer-disabled:active:after:border-gray-200 peer-focus:active:after:border-magenta-500 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100"
          for="test-1"
        >
          Option 2
        </label>
      </div>
    </div>
  </div>
</div>
`;
